### PR TITLE
releng - c7n-left docker image

### LINKED ISSUE #8216: shift-left - build into docker images
add c7n-left to default docker image, also consider a separate one as the default image is a bit large (cough, azure sdk)

### PR DESCRIPTION
**Pull request by kapilt**
c7n-left docker image, closes #8216 

trying out chainguard images which are smaller have some builtin sbom.

image size: 185mb

Screenshot 2023-03-16 at 11 57 32 AM

running
Screenshot 2023-03-19 at 7 51 31 AM

### CODE REVIEW

**Comment by kapilt** [comment]
using to dive to analyze, the image size is actually primarily due to 80mbs for botocore/boto and another 10mb for its dependencies, ie. a separate c7n_core would allow us to get image sizes ~100mb

**Comment by kapilt** [comment]
proposal on c7n_core is #5350

File: docker/c7n-left [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+RUN python -m venv $HOME/tools && \
+    source $HOME/tools/bin/activate && \
+    pip3 install poetry==${POETRY_VERSION}
+
+ADD tools/c7n_left /app
+
+RUN ~/tools/bin/poetry export -o requirements.txt
+RUN pip3 install --user -r requirements.txt
+RUN pip3 install --user c7n
```
Comment (kapilt) [inline docker/c7n-left:15]: ideally we'd install this out of repo, so it works the same as the other images, as opposed to picking latest release extant.

**Review by ajkerrigan — APPROVED** [review]
lgtm :tada:

**Pull request merged**